eCommerce-Application Sprint 1: Repository Setup, Project Management, CommerceTools Integration, and Development Environment Configuration

SvetaIlina opened the pull request:
1. Task: [link](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint%231.md)
2. Screenshot:
![image](https://github.com/makarovaiuliia/kickflip/assets/139857261/44c1df77-068d-4e58-81b0-0dbe8a8f8883)

3. Deploy: -
4. Done 06.05.2024 / deadline 07.05.2024
5. Score: 150 / 150
 - Repository Setup (29 points) 
  - [x] (10 points) Create a GitHub repository for the project, complete with a README file, .gitignore, and necessary dependencies. [RSS-ECOMM-1_01](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint1/RSS-ECOMM-1_01.md) 
- [x] (10 points) Set up a clear and organized folder structure within the repository, ensuring a maintainable and scalable codebase. [RSS-ECOMM-1_02](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint1/RSS-ECOMM-1_02.md) 
- [x] (4 points) Invite all team members and the mentor as collaborators, ensuring everyone has access to contribute to the repository. [RSS-ECOMM-1_03](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint1/RSS-ECOMM-1_03.md) 
- [x] (5 points) Create a pull request template with a clear structure for describing proposed changes and the rationale behind them. [RSS-ECOMM-1_21](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint1/RSS-ECOMM-1_21.md) 
 - Task Board Setup (Trello, Jira, etc.) (14 points) 
- [x] (6 points) Create a project task board using a project management tool (e.g., Trello or Jira), organizing tasks into lists or columns corresponding to their status (e.g., To Do, In Progress, Done). [RSS-ECOMM-1_04](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint1/RSS-ECOMM-1_04.md) 
- [x] (5 points) Break down project tasks into smaller, manageable subtasks and assign them to team members, ensuring a clear delegation of responsibilities. [RSS-ECOMM-1_05](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint1/RSS-ECOMM-1_05.md) 
- [x] (3 points) Share the task board with the team members and the mentor, ensuring everyone can track progress and update the status of their tasks. [RSS-ECOMM-1_18](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint1/RSS-ECOMM-1_18.md) 
 - CommerceTools Project and API Client Setup (30 points) 
- [x] (15 points) Create the project in CommerceTools, setting up the currencies, languages, and other configurations in the Merchant Center. [RSS-ECOMM-1_06](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint1/RSS-ECOMM-1_06.md) 
- [x] (15 points) Create the API client for CommerceTools, enabling access to the platform's APIs and managing permissions and scopes required for the project. [RSS-ECOMM-1_07](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint1/RSS-ECOMM-1_07.md) 
 - Development Environment Configuration (43 points) 
- [x] (10 points) Set up and configure a bundler (e.g., Webpack, Vite, Parcel, or other) for the project, supporting both development and production build modes. [RSS-ECOMM-1_08](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint1/RSS-ECOMM-1_08.md) 
- [x] (5 points) Configure TypeScript to enforce strong typing and improve code quality throughout the project. [RSS-ECOMM-1_09](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint1/RSS-ECOMM-1_09.md) 
- [x] (5 points) Set up ESLint to enforce consistent coding styles and identify potential issues in the codebase. [RSS-ECOMM-1_10](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint1/RSS-ECOMM-1_10.md) 
- [x] (5 points) Configure Prettier to automatically format code, ensuring a consistent and readable code style. [RSS-ECOMM-1_11](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint1/RSS-ECOMM-1_11.md) 
- [x] (8 points) Initialize Husky to manage Git hooks, automating tasks such as code formatting and linting checks during the commit process. [RSS-ECOMM-1_12](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint1/RSS-ECOMM-1_12.md) 
- [x] (10 points) Install and configure Jest or Vitest as the testing framework for the project, enabling comprehensive testing of code functionality. [RSS-ECOMM-1_13](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint1/RSS-ECOMM-1_13.md)
 - Development Scripts (15 points) 
- [x] (5 points) Create a script for running ESLint across the codebase, enabling quick and efficient identification of coding style issues and potential errors. [RSS-ECOMM-1_14](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint1/RSS-ECOMM-1_14.md) 
- [x] (5 points) Create a script for running Prettier across the codebase, simplifying code formatting and ensuring uniform code style. [RSS-ECOMM-1_15](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint1/RSS-ECOMM-1_15.md) 
- [x] (5 points) Create a script for running Jest or Vitest tests, streamlining the process of executing tests and generating reports on test outcomes. [RSS-ECOMM-1_16](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint1/RSS-ECOMM-1_16.md) 
 - Comprehensive README (19 points) 
- [x] (10 points) Add a thorough README file in the repository, describing the project, its purpose, and the technology stack used. [RSS-ECOMM-1_17](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint1/RSS-ECOMM-1_17.md) 
- [x] (5 points) Clearly document all available scripts (e.g., for running ESLint, Prettier, and Jest or Vitest tests) and their usage in the README file. [RSS-ECOMM-1_19](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint1/RSS-ECOMM-1_19.md) 
- [x] (4 points) Provide detailed instructions for setting up and running the project locally, ensuring newcomers and contributors can quickly get started. [RSS-ECOMM-1_20](https://github.com/rolling-scopes-school/tasks/blob/master/tasks/eCommerce-Application/Sprints/Sprint1/RSS-ECOMM-1_20.md)

makarovaiuliia committed:
feat: RSS-ECOMM-1_21 Create a pull request template

makarovaiuliia committed:
fix: RSS-ECOMM-1_21 rename pull_request_template.md

makarovaiuliia committed:
feat: RSS-ECOMM-1_02 Set up folder structure

makarovaiuliia committed:
feat: RSS-ECOMM-1_17 Update README with project description

SvetaIlina committed:
RSS-ECOMM-1 development environment configuration and scripts (#4)

* feat: RSS-ECOMM-1_08 Set up and configure Vite

* feat: RSS-ECOMM-1_09 Configure TypeScript

* refactor: RSS-ECOMM-1_08 remove unnecessary .gitignore

* feat: RSS-ECOMM-1_10 Set up ESLint

* feat: RSS-ECOMM-1_14 Create a script for running ESLint

* feat: RSS-ECOMM-1_11 Configure Prettier

* feat:   RSS-ECOMM-1_15 Create a script for running Prettier

* feat: RSS-ECOMM-1_12 Initialize Husky

* feat:  RSS-ECOMM-1_13 Install and configure Jest

* feat: RSS-ECOMM-1_16 Create a script for running Jest

* fix: add right config for ignore Patterns

* refactor: remove README

* feat: RSS-ECOMM-1_13 add updating jest config and new package

* feat: add new script for commits

SvetaIlina committed:
feat: add all available scripts (#5)

Pavel-Kuvshinov committed:
feat: RSS-ECOMM-1_06_07 Create commercetools project and API for commercetools project. (#6)

* feat: RSS-ECOMM-1_06 and RSS-ECOMM-1_07 Create comercetools project. Create API for commercetools project.

* fix: RSS-ECOMM-1_06 and RSS-ECOMM-1_07 Fix mistakes in Api env file.

SvetaIlina committed:
chore: delete duplicate folders and files, move files in to kickflip-project folder (#8)

SvetaIlina committed:
feat/RSS-ECOMM-2_01-02_Login_Page_Input_Validation (#9)

* style: add fonts

* style: add main styles

* feat: add login form component

* style: add loginForms styles

* feat: add functional for toggle passwords visibility

* feat: add email input validation and displaying errors

* style: add style for errors message and button hover

* feat: add password input validation and displaying errors

* feat: add button blocking in case of invalid data

* chore: delete duplicate folders and files, move files in to kickflip-project folder (#8)

* refactor: rename function and move it into separate folder

* feat: add loginForms component in to Login page

* chore: add new package

* test: add tests for loginForm

* style: change inputs and buttons style, add adaptive

* fix: make a predetermined minimum height for the error field

* style: add branded color for inputs text

* feat: remove special character password check

vladislavkovaliov commented on README.md:
````diff
@@ -0,0 +1,85 @@
+   ```
+3. **Start the development server:**
+   ```bash
+   npm start
````
я не вижу start скрипт в package.json, это точно верное описание?

makarovaiuliia replied:
Исправила

vladislavkovaliov commented on README.md:
````diff
@@ -0,0 +1,85 @@
+
+1. **Clone the repository:**
+   ```bash
+   git clone https://github.com/makarovaiuliia/kickflip.git
````
а зачем тут описание как клонировать репозиторий?)

makarovaiuliia replied:
Видела в реадми других такое)) думаешь лишнее?

vladislavkovaliov commented on kickFlip-project/kick-flip-v2-webstore-warrioirs_kick-flip-API.env:
```diff
@@ -0,0 +1,7 @@
+
+CTP_PROJECT_KEY=kick-flip_webstore-warriors
```
а как vite поймет что нужно прочитать этот .env файл?
по доке есть описание с приоритетами по названия для .env файлов
https://vitejs.dev/guide/env-and-mode#env-files

makarovaiuliia replied:
@Pavel-Kuvshinov

vladislavkovaliov commented on kickFlip-project/src/main.tsx:
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import ReactDOM from 'react-dom/client';
+import App from './App';
```
настроить alias было бы круто это в две-три строчки делается

makarovaiuliia replied:
Сделано

vladislavkovaliov commented on kickFlip-project/kick-flip-v2-webstore-warrioirs_kick-flip-API.env:
```diff
@@ -0,0 +1,7 @@
+
+CTP_PROJECT_KEY=kick-flip_webstore-warriors
+CTP_CLIENT_SECRET=N30vTyfozdhe1Wl5CHTmKrnNYgozBfMh
```
такие штуки не нужно в репу пушить их локально хранят, а для gitaction в secrets уносят

makarovaiuliia replied:
@Pavel-Kuvshinov

vladislavkovaliov commented:
**неправильная струтура папок** - желательно починить

makarovaiuliia commented:
Сделаем!

SvetaIlina committed:
chore: remove doubled folders

makarovaiuliia committed:
fix: Change script in readme file

SvetaIlina committed:
feat/RSS-ECOMM-2_09-10 registration page implementation/ Input Validation  (#10)

* style: add fonts

* style: add main styles

* feat: add login form component

* style: add loginForms styles

* feat: add functional for toggle passwords visibility

* feat: add email input validation and displaying errors

* style: add style for errors message and button hover

* feat: add password input validation and displaying errors

* feat: add button blocking in case of invalid data

* chore: delete duplicate folders and files, move files in to kickflip-project folder (#8)

* refactor: rename function and move it into separate folder

* feat: add loginForms component in to Login page

* chore: add new package

* test: add tests for loginForm

* style: change inputs and buttons style, add adaptive

* fix: make a predetermined minimum height for the error field

* feat: add new package

* refactor: rename files and add separate folders for forms

* feat: add enums for errors messages

* feat: add registration form

* feat: add registration page

* style: add styles for reg page

* refactor: remove unnecessary files after merging

* fix: change test after adding errors messages enums

* fix: password and zip-code validation

* refactor: change class name to a new one using a kebab-case

* refactor: corrected the spelling of the address property

* refactor: fix Codacy  issues

* style: add empty line before new rule

* style: add empty line

* style: add border and change styles for adaptive for reg form

* refactor: fix codacy erroes

makarovaiuliia committed:
feat: aliases in paths

makarovaiuliia committed:
Merge branch 'release/login-registration-main' into develop

makarovaiuliia committed:
fix: add .env file

makarovaiuliia committed:
refactor: diable eslint rule for alias

makarovaiuliia committed:
feat: Login, Registration and Main Pages Implementation

* style: all files

* fix: commit command add check

* chore: merge develop into release/login-registration-main branch

* chore: remove doubled folders

* fix: Change script in readme file

* feat: aliases in paths

---------

Co-authored-by: Svetlana <svetailina1987@gmail.com>

* feat/RSS-ECOMM-2_01-02 login page input validation (#11)

* style: add fonts

* style: add main styles

* feat: add login form component

* style: add loginForms styles

* feat: add functional for toggle passwords visibility

* feat: add email input validation and displaying errors

* style: add style for errors message and button hover

* feat: add password input validation and displaying errors

* feat: add button blocking in case of invalid data

* chore: delete duplicate folders and files, move files in to kickflip-project folder (#8)

* refactor: rename function and move it into separate folder

* feat: add loginForms component in to Login page

* chore: add new package

* test: add tests for loginForm

* style: change inputs and buttons style, add adaptive

* fix: make a predetermined minimum height for the error field

* style: add branded color for inputs text

* feat: remove special character password check

* feat: add separate function for login form resetting

* refactor: add formatting

* refactor: delete empty line

* refactor: remove empty line

* refactor: delete empty line

* Rss ecomm 2 01 02 login page input validation (#15)

* style: add fonts

* style: add main styles

* feat: add login form component

* style: add loginForms styles

* feat: add functional for toggle passwords visibility

* feat: add email input validation and displaying errors

* style: add style for errors message and button hover

* feat: add password input validation and displaying errors

* feat: add button blocking in case of invalid data

* chore: delete duplicate folders and files, move files in to kickflip-project folder (#8)

* refactor: rename function and move it into separate folder

* feat: add loginForms component in to Login page

* chore: add new package

* test: add tests for loginForm

* style: change inputs and buttons style, add adaptive

* fix: make a predetermined minimum height for the error field

* style: add branded color for inputs text

* feat: remove special character password check

* feat: add separate function for login form resetting

* refactor: add formatting

* refactor: delete empty line

* refactor: remove empty line

* refactor: delete empty line

* feat: update branch

* refactor: delete doubled file

* refactor: delete doubled file

* refactor: correct spelling

* feat: RSS ECOMM 2-03 Integrate login form with commerce tools authorization service

* chore: merge release/login-registration-main into RSS-ECOMM-02_03 (#13)

* feat/RSS-ECOMM-2_09-10 registration page implementation/ Input Validation  (#10)

* style: add fonts

* style: add main styles

* feat: add login form component

* style: add loginForms styles

* feat: add functional for toggle passwords visibility

* feat: add email input validation and displaying errors

* style: add style for errors message and button hover

* feat: add password input validation and displaying errors

* feat: add button blocking in case of invalid data

* chore: delete duplicate folders and files, move files in to kickflip-project folder (#8)

* refactor: rename function and move it into separate folder

* feat: add loginForms component in to Login page

* chore: add new package

* test: add tests for loginForm

* style: change inputs and buttons style, add adaptive

* fix: make a predetermined minimum height for the error field

* feat: add new package

* refactor: rename files and add separate folders for forms

* feat: add enums for errors messages

* feat: add registration form

* feat: add registration page

* style: add styles for reg page

* refactor: remove unnecessary files after merging

* fix: change test after adding errors messages enums

* fix: password and zip-code validation

* refactor: change class name to a new one using a kebab-case

* refactor: corrected the spelling of the address property

* refactor: fix Codacy  issues

* style: add empty line before new rule

* style: add empty line

* style: add border and change styles for adaptive for reg form

* refactor: fix codacy erroes

* chore: merge develop into release/login-registration-main branch

* chore: remove doubled folders

* fix: Change script in readme file

* feat: aliases in paths

---------

Co-authored-by: Svetlana <svetailina1987@gmail.com>

---------

Co-authored-by: SvetaIlina <139857261+SvetaIlina@users.noreply.github.com>
Co-authored-by: Svetlana <svetailina1987@gmail.com>

* feat: redux toolkit setting

* feat: implements userLogin api functionality

* fix: delete a file

* refactor: review comments

---------

Co-authored-by: SvetaIlina <139857261+SvetaIlina@users.noreply.github.com>
Co-authored-by: Svetlana <svetailina1987@gmail.com>

* refactor: delete spaces and add closing tag in form.css

* feat/RSS-ECOMM-2 create input field element (#19)

* feat: add inputField component

* feat: add FormField element

* feat/Rss ecomm 2 04 implement error handling for failed authentication attempts using commerce tools (#17)

* feat: add new package

* feat: add displaying authentication errors

* fix: displaying an error when the server responds

* feat: RSS-ECOMM 2_11/14/15_integrate registration form with commerce tools

* feat: signup api

* feat: add anonymous token flow

* feat: add error from registration in store

* feat: add shipping and billing addresses to the form

* fix: eslint correction

* fix: shippint city input

* fix: add form tag

* refactor: fix spelling mistake

* refactor: change input tags to input elements in form

* refactor: change Input tag to a component

* fix: add unique ids

* feat: add styles for regform and saveTokens function

* refactor: fix formField maltext

* style: change regForm column style

* feat: add refresh token functionality

* refactor: fix eslint error

* feat: add getUser functionality

* feat/RSS-ECOMM-2_12 implement error handling for failed registration attempts (#22)

* style: change columns size

* feat: add displaying error for failed registration attempts

* fix: postal code error display when it is filled in before country

* feat: add auth with refreshToken and login with accessToken

* feat: RSS-ECOMM-2-08-16-17-18-19-20-21 Add page navigation with basic routing

* feat: RSS-ECOMM-2_17-18-19 Add basic pages navigation. Add Basic Routing.

* feat: RSS-ECOMM-2_20 Add "404" page.

* feat: RSS-ECOMM-2_08-16-21 Add navigation to login and registration pages for unauthorized users.

* fix: RSS-ECOMM-2_17-18-19 A separate "Header" component has been added. A "basicLayout" component has been added. Links to the "Log in" and "Cart" pages has been added. Some mistakes in styles has been fixed.

* feat: refactor, style redirect promt, add NavLink tag

* style: active links

---------

Co-authored-by: Iuliia Makarova <iuliia.makarova.au@gmail.com>
Co-authored-by: iuliia <144937274+makarovaiuliia@users.noreply.github.com>

* refactor: remove already declared module statement

* fix: sending invalid form when pressing enter (#35)

* refactor: add .env file to .gitignore

* feat: user login using tokens and routing features

* merge Release/login registration main into branch userLoginUsingTokens

* feat/RSS-ECOMM-2_12 implement error handling for failed registration attempts (#22)

* style: change columns size

* feat: add displaying error for failed registration attempts

* fix: postal code error display when it is filled in before country

* feat: RSS-ECOMM-2-08-16-17-18-19-20-21 Add page navigation with basic routing

* feat: RSS-ECOMM-2_17-18-19 Add basic pages navigation. Add Basic Routing.

* feat: RSS-ECOMM-2_20 Add "404" page.

* feat: RSS-ECOMM-2_08-16-21 Add navigation to login and registration pages for unauthorized users.

* fix: RSS-ECOMM-2_17-18-19 A separate "Header" component has been added. A "basicLayout" component has been added. Links to the "Log in" and "Cart" pages has been added. Some mistakes in styles has been fixed.

* feat: refactor, style redirect promt, add NavLink tag

* style: active links

---------

Co-authored-by: Iuliia Makarova <iuliia.makarova.au@gmail.com>
Co-authored-by: iuliia <144937274+makarovaiuliia@users.noreply.github.com>

---------

Co-authored-by: SvetaIlina <139857261+SvetaIlina@users.noreply.github.com>
Co-authored-by: Pavel-Kuvshinov <94765270+Pavel-Kuvshinov@users.noreply.github.com>

* feat: add redirection after successful registration and login

* fix: token in authorization headers

* feat: isAuth dependancy in navLink

* refactor: turn paths to aliases

* feat: add protected routes

* feat: add logout functionality

* fix: bug with deleting tokens after logout

* feat: add shipping and billing addresses to the customer profile

* feat: add env file and remove eslint rule

* refactor: add basic token from .env file

---------

Co-authored-by: SvetaIlina <139857261+SvetaIlina@users.noreply.github.com>
Co-authored-by: Pavel-Kuvshinov <94765270+Pavel-Kuvshinov@users.noreply.github.com>

* refactor: remove dublicates

* fix: RSS-ECOMM-2_17-Add responsive styles for navigation. Add favicon. (#36)

* merge Release/login registration main into branch userLoginUsingTokens

* feat/RSS-ECOMM-2_12 implement error handling for failed registration attempts (#22)

* style: change columns size

* feat: add displaying error for failed registration attempts

* fix: postal code error display when it is filled in before country

* feat: RSS-ECOMM-2-08-16-17-18-19-20-21 Add page navigation with basic routing

* feat: RSS-ECOMM-2_17-18-19 Add basic pages navigation. Add Basic Routing.

* feat: RSS-ECOMM-2_20 Add "404" page.

* feat: RSS-ECOMM-2_08-16-21 Add navigation to login and registration pages for unauthorized users.

* fix: RSS-ECOMM-2_17-18-19 A separate "Header" component has been added. A "basicLayout" component has been added. Links to the "Log in" and "Cart" pages has been added. Some mistakes in styles has been fixed.

* feat: refactor, style redirect promt, add NavLink tag

* style: active links

---------

Co-authored-by: Iuliia Makarova <iuliia.makarova.au@gmail.com>
Co-authored-by: iuliia <144937274+makarovaiuliia@users.noreply.github.com>

---------

Co-authored-by: SvetaIlina <139857261+SvetaIlina@users.noreply.github.com>
Co-authored-by: Pavel-Kuvshinov <94765270+Pavel-Kuvshinov@users.noreply.github.com>

* feat: add redirection after successful registration and login

* fix: token in authorization headers

* feat: isAuth dependancy in navLink

* refactor: turn paths to aliases

* feat: add protected routes

* feat: add logout functionality

* fix: bug with deleting tokens after logout

* feat: add shipping and billing addresses to the customer profile

* feat: add env file and remove eslint rule

* refactor: add basic token from .env file

* feat: RSS-ECOMM-2_17 Add responsive header styles.

* feat: RSS-ECOMM-2_17_AddResponsiveHeader Add faviicon. Fix mistake in log out button click function.

* fix: RSS-ECOMM-2_17_AddResponsiveHeader Add nessary rull for favicon in vite.config.ts.

* fix: RSS-ECOMM-2_17_AddResponsiveHeader fix naming destinations. Fix some responsive styles in header.

---------

Co-authored-by: iuliia <144937274+makarovaiuliia@users.noreply.github.com>
Co-authored-by: SvetaIlina <139857261+SvetaIlina@users.noreply.github.com>
Co-authored-by: Iuliia Makarova <iuliia.makarova.au@gmail.com>

* Fix: RSS ECOMM 2-09-10 Display clear error messages for invalid inputs in the registration form

* fix: change regex for a proper check

* fix: email regexp in login form

* feat: RSS-ECOMM-2_11 displaying successful registration message

* feat: add new user.state properties

* feat: add success message  to the user upon successful account creation

* style: check style before build

* Rename notfoundpage to tempname

* Rename tempname to notFoundPage

* change notFoundPage to tempnames

* fix: change tempname to notFoundPage

* feat: routing setting Netlify

* refactor: move assets folder to public directory

* refactor: put icons in public directory

* refactor: move fonts to src directory

* fix: font paths in main.css

* fix: import of svg files in deploy

* refactor: change paths

* fix: fonts file

* refactor: paths

* fix: typo in path

---------

Co-authored-by: Svetlana <svetailina1987@gmail.com>
Co-authored-by: SvetaIlina <139857261+SvetaIlina@users.noreply.github.com>
Co-authored-by: Pavel-Kuvshinov <94765270+Pavel-Kuvshinov@users.noreply.github.com>

makarovaiuliia committed:
feat: Catalog Product Page, Detailed Product Page & User Profile Page Implementation

* fix: RSS-ECOMM-3_0 Fix some styles in sections (#43)

* style: all files

* fix: commit command add check

* chore: merge develop into release/login-registration-main branch

* chore: remove doubled folders

* fix: Change script in readme file

* feat: aliases in paths

---------

Co-authored-by: Svetlana <svetailina1987@gmail.com>

* feat/RSS-ECOMM-2_01-02 login page input validation (#11)

* style: add fonts

* style: add main styles

* feat: add login form component

* style: add loginForms styles

* feat: add functional for toggle passwords visibility

* feat: add email input validation and displaying errors

* style: add style for errors message and button hover

* feat: add password input validation and displaying errors

* feat: add button blocking in case of invalid data

* chore: delete duplicate folders and files, move files in to kickflip-project folder (#8)

* refactor: rename function and move it into separate folder

* feat: add loginForms component in to Login page

* chore: add new package

* test: add tests for loginForm

* style: change inputs and buttons style, add adaptive

* fix: make a predetermined minimum height for the error field

* style: add branded color for inputs text

* feat: remove special character password check

* feat: add separate function for login form resetting

* refactor: add formatting

* refactor: delete empty line

* refactor: remove empty line

* refactor: delete empty line

* Rss ecomm 2 01 02 login page input validation (#15)

* style: add fonts

* style: add main styles

* feat: add login form component

* style: add loginForms styles

* feat: add functional for toggle passwords visibility

* feat: add email input validation and displaying errors

* style: add style for errors message and button hover

* feat: add password input validation and displaying errors

* feat: add button blocking in case of invalid data

* chore: delete duplicate folders and files, move files in to kickflip-project folder (#8)

* refactor: rename function and move it into separate folder

* feat: add loginForms component in to Login page

* chore: add new package

* test: add tests for loginForm

* style: change inputs and buttons style, add adaptive

* fix: make a predetermined minimum height for the error field

* style: add branded color for inputs text

* feat: remove special character password check

* feat: add separate function for login form resetting

* refactor: add formatting

* refactor: delete empty line

* refactor: remove empty line

* refactor: delete empty line

* feat: update branch

* refactor: delete doubled file

* refactor: delete doubled file

* refactor: correct spelling

* feat: RSS ECOMM 2-03 Integrate login form with commerce tools authorization service

* chore: merge release/login-registration-main into RSS-ECOMM-02_03 (#13)

* feat/RSS-ECOMM-2_09-10 registration page implementation/ Input Validation  (#10)

* style: add fonts

* style: add main styles

* feat: add login form component

* style: add loginForms styles

* feat: add functional for toggle passwords visibility

* feat: add email input validation and displaying errors

* style: add style for errors message and button hover

* feat: add password input validation and displaying errors

* feat: add button blocking in case of invalid data

* chore: delete duplicate folders and files, move files in to kickflip-project folder (#8)

* refactor: rename function and move it into separate folder

* feat: add loginForms component in to Login page

* chore: add new package

* test: add tests for loginForm

* style: change inputs and buttons style, add adaptive

* fix: make a predetermined minimum height for the error field

* feat: add new package

* refactor: rename files and add separate folders for forms

* feat: add enums for errors messages

* feat: add registration form

* feat: add registration page

* style: add styles for reg page

* refactor: remove unnecessary files after merging

* fix: change test after adding errors messages enums

* fix: password and zip-code validation

* refactor: change class name to a new one using a kebab-case

* refactor: corrected the spelling of the address property

* refactor: fix Codacy  issues

* style: add empty line before new rule

* style: add empty line

* style: add border and change styles for adaptive for reg form

* refactor: fix codacy erroes

* chore: merge develop into release/login-registration-main branch

* chore: remove doubled folders

* fix: Change script in readme file

* feat: aliases in paths

---------

Co-authored-by: Svetlana <svetailina1987@gmail.com>

---------

Co-authored-by: SvetaIlina <139857261+SvetaIlina@users.noreply.github.com>
Co-authored-by: Svetlana <svetailina1987@gmail.com>

* feat: redux toolkit setting

* feat: implements userLogin api functionality

* fix: delete a file

* refactor: review comments

---------

Co-authored-by: SvetaIlina <139857261+SvetaIlina@users.noreply.github.com>
Co-authored-by: Svetlana <svetailina1987@gmail.com>

* refactor: delete spaces and add closing tag in form.css

* feat/RSS-ECOMM-2 create input field element (#19)

* feat: add inputField component

* feat: add FormField element

* feat/Rss ecomm 2 04 implement error handling for failed authentication attempts using commerce tools (#17)

* feat: add new package

* feat: add displaying authentication errors

* fix: displaying an error when the server responds

* feat: RSS-ECOMM 2_11/14/15_integrate registration form with commerce tools

* feat: signup api

* feat: add anonymous token flow

* feat: add error from registration in store

* feat: add shipping and billing addresses to the form

* fix: eslint correction

* fix: shippint city input

* fix: add form tag

* refactor: fix spelling mistake

* refactor: change input tags to input elements in form

* refactor: change Input tag to a component

* fix: add unique ids

* feat: add styles for regform and saveTokens function

* refactor: fix formField maltext

* style: change regForm column style

* feat: add refresh token functionality

* refactor: fix eslint error

* feat: add getUser functionality

* feat/RSS-ECOMM-2_12 implement error handling for failed registration attempts (#22)

* style: change columns size

* feat: add displaying error for failed registration attempts

* fix: postal code error display when it is filled in before country

* feat: add auth with refreshToken and login with accessToken

* feat: RSS-ECOMM-2-08-16-17-18-19-20-21 Add page navigation with basic routing

* feat: RSS-ECOMM-2_17-18-19 Add basic pages navigation. Add Basic Routing.

* feat: RSS-ECOMM-2_20 Add "404" page.

* feat: RSS-ECOMM-2_08-16-21 Add navigation to login and registration pages for unauthorized users.

* fix: RSS-ECOMM-2_17-18-19 A separate "Header" component has been added. A "basicLayout" component has been added. Links to the "Log in" and "Cart" pages has been added. Some mistakes in styles has been fixed.

* feat: refactor, style redirect promt, add NavLink tag

* style: active links

---------

Co-authored-by: Iuliia Makarova <iuliia.makarova.au@gmail.com>
Co-authored-by: iuliia <144937274+makarovaiuliia@users.noreply.github.com>

* refactor: remove already declared module statement

* fix: sending invalid form when pressing enter (#35)

* refactor: add .env file to .gitignore

* feat: user login using tokens and routing features

* merge Release/login registration main into branch userLoginUsingTokens

* feat/RSS-ECOMM-2_12 implement error handling for failed registration attempts (#22)

* style: change columns size

* feat: add displaying error for failed registration attempts

* fix: postal code error display when it is filled in before country

* feat: RSS-ECOMM-2-08-16-17-18-19-20-21 Add page navigation with basic routing

* feat: RSS-ECOMM-2_17-18-19 Add basic pages navigation. Add Basic Routing.

* feat: RSS-ECOMM-2_20 Add "404" page.

* feat: RSS-ECOMM-2_08-16-21 Add navigation to login and registration pages for unauthorized users.

* fix: RSS-ECOMM-2_17-18-19 A separate "Header" component has been added. A "basicLayout" component has been added. Links to the "Log in" and "Cart" pages has been added. Some mistakes in styles has been fixed.

* feat: refactor, style redirect promt, add NavLink tag

* style: active links

---------

Co-authored-by: Iuliia Makarova <iuliia.makarova.au@gmail.com>
Co-authored-by: iuliia <144937274+makarovaiuliia@users.noreply.github.com>

---------

Co-authored-by: SvetaIlina <139857261+SvetaIlina@users.noreply.github.com>
Co-authored-by: Pavel-Kuvshinov <94765270+Pavel-Kuvshinov@users.noreply.github.com>

* feat: add redirection after successful registration and login

* fix: token in authorization headers

* feat: isAuth dependancy in navLink

* refactor: turn paths to aliases

* feat: add protected routes

* feat: add logout functionality

* fix: bug with deleting tokens after logout

* feat: add shipping and billing addresses to the customer profile

* feat: add env file and remove eslint rule

* refactor: add basic token from .env file

---------

Co-authored-by: SvetaIlina <139857261+SvetaIlina@users.noreply.github.com>
Co-authored-by: Pavel-Kuvshinov <94765270+Pavel-Kuvshinov@users.noreply.github.com>

* refactor: remove dublicates

* fix: RSS-ECOMM-2_17-Add responsive styles for navigation. Add favicon. (#36)

* merge Release/login registration main into branch userLoginUsingTokens

* feat/RSS-ECOMM-2_12 implement error handling for failed registration attempts (#22)

* style: change columns size

* feat: add displaying error for failed registration attempts

* fix: postal code error display when it is filled in before country

* feat: RSS-ECOMM-2-08-16-17-18-19-20-21 Add page navigation with basic routing

* feat: RSS-ECOMM-2_17-18-19 Add basic pages navigation. Add Basic Routing.

* feat: RSS-ECOMM-2_20 Add "404" page.

* feat: RSS-ECOMM-2_08-16-21 Add navigation to login and registration pages for unauthorized users.

* fix: RSS-ECOMM-2_17-18-19 A separate "Header" component has been added. A "basicLayout" component has been added. Links to the "Log in" and "Cart" pages has been added. Some mistakes in styles has been fixed.

* feat: refactor, style redirect promt, add NavLink tag

* style: active links

---------

Co-authored-by: Iuliia Makarova <iuliia.makarova.au@gmail.com>
Co-authored-by: iuliia <144937274+makarovaiuliia@users.noreply.github.com>

---------

Co-authored-by: SvetaIlina <139857261+SvetaIlina@users.noreply.github.com>
Co-authored-by: Pavel-Kuvshinov <94765270+Pavel-Kuvshinov@users.noreply.github.com>

* feat: add redirection after successful registration and login

* fix: token in authorization headers

* feat: isAuth dependancy in navLink

* refactor: turn paths to aliases

* feat: add protected routes

* feat: add logout functionality

* fix: bug with deleting tokens after logout

* feat: add shipping and billing addresses to the customer profile

* feat: add env file and remove eslint rule

* refactor: add basic token from .env file

* feat: RSS-ECOMM-2_17 Add responsive header styles.

* feat: RSS-ECOMM-2_17_AddResponsiveHeader Add faviicon. Fix mistake in log out button click function.

* fix: RSS-ECOMM-2_17_AddResponsiveHeader Add nessary rull for favicon in vite.config.ts.

* fix: RSS-ECOMM-2_17_AddResponsiveHeader fix naming destinations. Fix some responsive styles in header.

---------

Co-authored-by: iuliia <144937274+makarovaiuliia@users.noreply.github.com>
Co-authored-by: SvetaIlina <139857261+SvetaIlina@users.noreply.github.com>
Co-authored-by: Iuliia Makarova <iuliia.makarova.au@gmail.com>

* Fix: RSS ECOMM 2-09-10 Display clear error messages for invalid inputs in the registration form

* fix: change regex for a proper check

* fix: email regexp in login form

* feat: RSS-ECOMM-2_11 displaying successful registration message

* feat: add new user.state properties

* feat: add success message  to the user upon successful account creation

* style: check style before build

* Rename notfoundpage to tempname

* Rename tempname to notFoundPage

* change notFoundPage to tempnames

* fix: change tempname to notFoundPage

* feat: routing setting Netlify

* refactor: move assets folder to public directory

* refactor: put icons in public directory

* refactor: move fonts to src directory

* fix: font paths in main.css

* fix: import of svg files in deploy

* refactor: change paths

* fix: fonts file

* refactor: paths

* fix: typo in path

* fix: RSS-ECOMM-3_0 Fix some styles in Header, "Main" section, "Login" page and "Registration" page.

* feat: RSS-ECOMM-3 Add separated wrapper classes for main section.

* fix: RSS-ECOMM-3 Add separated wrapper classes for main section on all pages.

* fix: RSS-ECOMM-3 deleted unnessary file.

---------

Co-authored-by: Iuliia Makarova <iuliia.makarova.au@gmail.com>
Co-authored-by: iuliia <144937274+makarovaiuliia@users.noreply.github.com>
Co-authored-by: Svetlana <svetailina1987@gmail.com>
Co-authored-by: SvetaIlina <139857261+SvetaIlina@users.noreply.github.com>

* feat: RSS-ECOMM-3-06/07 Interactive product cards, Redirect to detailed product page (#44)

* feat: add types for Product Response

* feat: add Product card design

* feat: add hover card functionality

* refactor: css file

* feat: add route for the detailed product page

* refactor: add slug to params

* refactor: getImageFromEachColor function

* feat: add comment about path

* RSS-ECOMM-3-20 Profile page routing (#46)

* feat: RSS-ECOMM-3_20-Add profile page routing. Add main links with styles without responsive styles.

* fix: RSS-ECOMM-3_20 Delete unnessary classes from the page.

* feat: RSS-ECOMM-3_20 fix profile styles.

* fix: server connection (#47)

* fix: add login request after registration

* fix: message in error when refreshing

* refactor: turn query parameters into body

* feat/RSS-ECOMM-3_09: Fetch and Display Product Details using Commercetools API (#49)

* feat: add function for getting product

* feat: add new components: SideImages and MainImage

* feat: add general component for displayind products images

* feat: add a separate type for the return value from the processVariants function

* feat: add VariantsImages component

* refactor: update imports

* feat: add product Description and Info components

* feat: add ProductSizes component

* feat: add general component DetailsContainer for displaying products info

* feat: update Product component with new components

* style: remove unnecessary font

* feat: RSS ECOMM 3-01/02 Fetch and display product list & Display prices with discount (#48)

* feat: add product get request

* feat: change image functionality

* style: fix main wrapper

* feat: add discounted price

* feat: add card description

* fix: delete mock data

* feat: change euro to dollar

* fix: review comments

* fix: euro to dollar

* refactor: adjustments

* fix: errors after merge

* fix: delete s from component

* style: change fonts

* style: add separate @font-face for Clash Display weight, add right font-family for forms elements

* feat: RSS-ECOMM-3-08 Implement category and breadcrumb navigation (#51)

* feat: add breadcrumbs component

* style: update

* feat: adds category navigation

* refactor: change naming

* fix: add async function when fetching token and products

* fix: error after merge

* Feat/RSS-ECOMM-3_09  (#50)

* style: add styles making header and footer stick to page edges

* feat: add route to product page

* feat: add Lodera like a separate element

* fix: delete unnecessary line

* feat: add dynamic loader

* feat: add  align-items

* style: changes in style

* merge: RSS-ECOMM-3_09 

* style: add styles making header and footer stick to page edges

* feat: add route to product page

* feat: add Lodera like a separate element

* fix: delete unnecessary line

* feat: add dynamic loader

* feat: add  align-items

Co-authored-by: SvetaIlina <139857261+SvetaIlina@users.noreply.github.com>

* feat: add accordeon components

* feat: accordeon list

* feat: add/remove category functionality

* style: styles for button

* feat: add ProductPrices component (#53)

* feat/RSS-ECOMM-3_10: Implement an Image Slider for Product Images  (#55)

* feat: add slider functionality for main image

* style: add styles

* feat: add filter category functionality

* fix: selected categories

* feat: add useEffect for sneakers

* refactor: change product endpoint to product-projection

* refactor: change product endpoint to product-projection

* fix: config files

* feat: add category seatcj

* feat: add sorting functionality

* refactor: refactor sort params

* feat: add search functionality

* feat: add select color functionality and setImage

* fix: setImage functionality

* fix: color filter with useMemo

* refactor: small

* feat/RSS-ECOMM-3_12-13 Implement Enlarged Image Modal  whith Slider (#56)

* feat: add ModalWindow component

* feat: add modal slider

* style: add control-buttons container, move close btn

* feat: add adaptive disign

* style: change modal window animation and backgraund

* feat: add closing the modal window by clicking on the overlay and press Esc

* refactor: codacy checks

* refactor: change placeholder to search

* style: add hover to filter options

* fix: change tag a to link

* style: add circle near color name

* feat: add dollar sign to a price

* feat: add error filtering

* style: change style for selected filter options

* style: change color for filter options

* feat: implements search with the use of the server

* refactor: change layout in card list component

* Revert "RSS-ECOMM_3_03/04/05 Implement filter sort&search" (#74)

* feat: RSS-ECOMM_3_03/04/05 implement filter sort&search (#75)

* merge release into the the branch

* Feat/RSS-ECOMM-3_09  (#50)

* style: add styles making header and footer stick to page edges

* feat: add route to product page

* feat: add Lodera like a separate element

* fix: delete unnecessary line

* feat: add dynamic loader

* feat: add  align-items

* feat: add ProductPrices component (#53)

* feat/RSS-ECOMM-3_10: Implement an Image Slider for Product Images  (#55)

* feat: add slider functionality for main image

* style: add styles

* feat/RSS-ECOMM-3_12-13 Implement Enlarged Image Modal  whith Slider (#56)

* feat: add ModalWindow component

* feat: add modal slider

* style: add control-buttons container, move close btn

* feat: add adaptive disign

* style: change modal window animation and backgraund

* feat: add closing the modal window by clicking on the overlay and press Esc

* refactor: codacy checks

---------

Co-authored-by: SvetaIlina <139857261+SvetaIlina@users.noreply.github.com>

* refactor: style and elements

* feat: add adaptive design

* feat: close filters when mobile

* refactor: search depending on isMobile

* refactor: сщву

* fix: change sneakers to catalog

* refactor: change icon to text in mobile version

---------

Co-authored-by: SvetaIlina <139857261+SvetaIlina@users.noreply.github.com>

* refactor: change images path

* feat: add breadcrumbs to product page and add outlet section with routing

* feat: add breadcrumbs to product page

* feat: add outlet section

* fix: click on filter span bag

* feat: add outlet routing

* fix: build fix

* style: add adaptive design

* fix: breadcrumbs on product page

* feat: add a routing for outlet

* fix: outlet routing

* refactor: add outlet page

* Fix elements displaying (#78)

* feat: add loader wrapper for correct displaying

* fix: displaying breadCrumbs after productData loading

* RSS-ECOMM-3_14-15 Present the user's personal information (#54)

* feat: RSS-ECOMM-3_14 Add profile account detailed information with layout. Add change password layout.

* feat: RSS-ECOMM-3_14 Add addresses account information with layout. Add responsive styles. Add edit mode.

* fix: RSS-ECOMM-3_14-15 Fix mistakes on the change password section. Add change password abillity.

* fix: RSS-ECOMM-3_14-15 Fix mistakes in types and functions jn the profile user data change. Fix mistake in routing to the profile page after sign up user.

* fix: RSS-ECOMM-3_14-15 Fix mistakes in types and functions jn the profile user address change. Fix mistake in edit function on the address and user data forms.

* fix: RSS-ECOMM-3_14-15 Fix mistakes in the user data update form.

* refactor: RSS-ECOMM-3_14-15 Refactor code.

* fix: RSS-ECOMM-3_14-15 Add cansel edit mode function. Fix some styles for errors messages in input fields.

* fix: RSS-ECOMM-3_14-15 Fix class naming in shiiping or billing forms on the profile page.

* fix: RSS-ECOMM-3_14-15 Remove console.logs.

* fix: RSS-ECOMM-3_16 Add change user data ability. Add modal with message. Fix some mistakes in code.

* feat: RSS-ECOMM-3_18 Add changing user address ability mith modal.

* fix: RSS-ECOMM-3_18 Fix some mistakes in styles.

* fix: RSS-ECOMM-3_18 Added ability to create new user address.

* fix: RSS-ECOMM-3_18 Added ability to delete user address.

* fix: RSS-ECOMM-3_18 Added ability to select te default user address item.

* refactor: RSS-ECOMM-3_14 Refactor address updating section on the profile page.

* fix: RSS-ECOMM-3_18 Add cancel abillity for adding new address.

* fix: RSS-ECOMM-3_16 Add abillity to change default billing and shipping addresses.

* fix: RSS-ECOMM-3_14-15-16-17-18-19-20-21-22 Fix many mistakes in code.

* fix: RSS-ECOMM-3_18 Fix label headings.

* fix: RSS-ECOMM-3_17-18 Fix some mistakes.

* fix: build

* refactor: products -> catalog

* fix: RSS-ECOMM-3_14 Fix styles in the profile page. (#79)

* fix: correct selected filter on outlet page

---------

Co-authored-by: Pavel-Kuvshinov <94765270+Pavel-Kuvshinov@users.noreply.github.com>
Co-authored-by: Svetlana <svetailina1987@gmail.com>
Co-authored-by: SvetaIlina <139857261+SvetaIlina@users.noreply.github.com>

makarovaiuliia committed:
feat: footer and main page implementation (#84)

* feat: add footer design

* style: add responsive design

* feat: RSS-ECOMM-04_01 Add Add to Cart Button on Product Cards

* feat: add add to cart button

* feat: change button text when in the cart

* fix: add base tag to index.html for routing

* feat: add hero and benefits sections to main page

* feat: add menu section and promocode

* feat: RSS-ECOMM-4_03 Implement Efficient Loading of Products on Catalog Page (#81)

* feat: add category in search query

* feat: add infinite scroll

* fix: filter data variable

* feat: use redux for infinite scroll

* refactor: review comments

* RSS-ECOMM-4_10: Fetch and Display Cart Items (#85)

* feat: add cart type

* feat: add getFormatPrice function

* feat: add types for getting cart response

* feat: add apis function for getting carts data/ add mockCartData

* feat: add carts components

* fix: change styles and add description instead size/color attributes

* feat: RSS-ECOMM-4_19-20-22-24 Add About Us page and etc. (#86)

* feat: RSS-ECOMM-4_19-20-21-22-23-24 Add About Us page with routing. Add team members description with photos. Add rsschool link with icon. Fix styles in header navigation.

* fix: RSS-ECOMM-4_19-20-21-22-23-24 Add responsive styles to the About us page.

* feat: RSS ECOMM 4 02 integrate catalog page with commercetools api

* feat: add customer id field in slice

* fix: catalog rendering

* fix: build failure

* feat: add cart slice and card size form

* feat: add get variant id

* feat: add signIn through me endpoint

* feat: add server connection

* fix: add to cart request

* feat: add merge carts functionality

* feat: add cart items in redux

* fix: get active cart instead of all carts

* fix: add version into mock cart data

* fix: review comments

* fix: review

* style: add styling for card list

* style: remove media for card form

---------

Co-authored-by: SvetaIlina <139857261+SvetaIlina@users.noreply.github.com>
Co-authored-by: Pavel-Kuvshinov <94765270+Pavel-Kuvshinov@users.noreply.github.com>

makarovaiuliia committed:
Revert "feat: footer and main page implementation (#84)" (#90)

This reverts commit aba728f933f85c774843552db568042d4ea4779c.

makarovaiuliia committed:
fix: breadcrumbs

makarovaiuliia committed:
Update README.md

makarovaiuliia committed:
Update README.md

makarovaiuliia committed:
Update README.md
